File: ios/Sources/EmarsysPlugin/EmarsysPush.swift
import Foundation
import EmarsysSDK

@objc public class EmarsysPush: NSObject {
    @objc public func setPushToken(pushToken: String, completion: @escaping (Error?) -> Void) {
        let tokenData = Data(hexString: pushToken)
        Emarsys.push.setPushToken(pushToken: tokenData) { error in
            completion(error)
        }
    }

    @objc public func clearPushToken(completion: @escaping (Error?) -> Void) {
        Emarsys.push.clearPushToken() { error in
            completion(error)
        }
    }

    @objc public func getPushToken() -> String? {
        guard let tokenData = Emarsys.push.pushToken() else { return nil }
        return tokenData.hexString
    }
}

File: android/src/main/java/com/emarsys/plugin/emarsys/EmarsysPush.java
package com.emarsys.plugin.emarsys;

import com.emarsys.Emarsys;

public class EmarsysPush {

    public void setPushToken(String pushToken, EmarsysCompletionListener listener) {
        Emarsys.getPush().setPushToken(pushToken, (error) -> listener.onComplete(error));
    }

    public void clearPushToken(EmarsysCompletionListener listener) {
        Emarsys.getPush().clearPushToken((error) -> listener.onComplete(error));
    }

    public String getPushToken() {
        return Emarsys.getPush().getPushToken();
    }
}
